### PR TITLE
DOCKER-262 Add jdk21 to jar-runner

### PR DESCRIPTION
**Pull request by allen-ziegenfus**
- **DOCKER-258 - Revert DOCKER-257**
- **LPD-32411 Publish distro BOM .pom file when publishing a new release**
- **LPD-32411 Publish distro BOM .jar file when publishing a new release**
- **LPD-32411 Add test to cover promotion of distro BOMs**
- **LPD-32410 Fixing issue where the previous release should not keep the promoted == true**
- **LPD-32410 Adding test for process_new_product**
- **LPD-32410 Sort**
- **LPD-24539 Move invoke_github_api to _github.sh**
- **LPD-24539 Create test for GitHub tag creation**
- **LPD-24539 Add script to run all tests**
- **LPD-24539 Create a GitHub Action to run tests related to release scripts**
- **LPD-24539 Cover the possibility of a fail assertion among success assertions**
- **LPD-24539 Skip the test if liferay-portal-ee does not exist**
- **LPD-24539 Wordsmith**
- **LPD-24539 Add local**
- **LPD-24539 Rename**
- **2024.q2.10**
- **7.0.4 change log**
- **LPD-31859 Not using the cached file**
- **LPD-31859 Fix typo**
- **LPD-34371 Fixing the target platform version on release.properties for ga releases**
- **LPD-34371 Adding test for generate_release_properties_file**
- **u126**
- **2024.q2.11**
- **LPD-32410 Skip cache for download file**
- **LPD-32530 Clean logs folder**
- **LPD-31467 Validate the type of artifact before trying to download it**
- **LPD-31467 Check if folder exists**
- **LPD-31467 Remove duplication**
- **DOCKER-259 Batch image script fails when payload size is too big**
- **DOCKER-259 Rename and hard code it**
- **7.0.5 change log**
- **LPD-34563 Fix issue that prevented the script from entering the if condition**
- **LPD-34563 Add the releases content to GCP**
- **LPD-34563 More an info**
- **Bump nightlies to 4 hours for now**
- **SRE-6161 The 10_after_import.sql script should also delete the Analytics Cloud configuration**
- **SRE-6161 Delete the MFA settings in the same way it is done in the MODL script https://github.com/liferay/liferay-lxc/blob/8272ce1583802703248359a2018b87c8d149386e/liferay/configs/common/scripts/05-modl-sql-scripts.sh#L75**
- **SRE-6161 As of 2024.Q2, 'Configuration_' is no longer a control table, so 10_after_import.sql must delete data from all DB partitions**
- **SRE-6161 Sort**
- **2024.q2.12**
- **LRP-4868 Making the curls errors more meaningful**
- **LRP-4868 Changing the execution order of the release_gold**
- **LRP-4868 SF**
- **Revert "Bump nightlies to 4 hours for now"**
- **SRE-6187 Migrate Spinner to liferay-lxc repository**
- **SRE-6187 Not needed**
- **Add support to 2024.q3**
- **DOCKER-256 Set the JDK 21 as the default JDK shipped in bundles**
- **DOCKER-256 Starting from release 2024.q3, set JDK 21 as the default JDK shipped in bundles.**
- **DOCKER-256 Force recreation of Elasticsearch files when using JDK 21**
- **DOCKER-256 Move _test_common.sh to reuse out of release folder**
- **DOCKER-256 Prepare build_bundle_image.sh for tests**
- **DOCKER-256 Add tests**
- **DOCKER-256 Add to test suite tests out of release folder**
- **DOCKER-256 Rename to improve clarity**
- **DOCKER-256 SF**
- **DOCKER-256 Sublime sort**
- **DOCKER-256 SF to improve tests output reading**
- **LRP-4868 Restore check_url behavior**
- **2024.q3.0**
- **7.0.6 change log**
- **LPD-36284 Comply with OSGI specifications when generating GA distro jars**
- **LPD-36284 SF**
- **7.4.3.125-ga125**
- **LPD-34925 Print which test file is running**
- **2024.q3.1**
- **LPD-37190 Update release workflow**
- **LPD-37190 Removing previous releases candidates**
- **2024.q3.2**
- **LPD-30881 Fixing the order of the cache clean-up snippet**
- **LPD-30881 Just do it?**
- **DOCKER-260 Enable Ubuntu Pro attach #minorchange**
- **DOCKER-260 SF**
- **DOCKER-260 Make it one**
- **DOCKER-260 SF**
- **7.1.0 change log**
- **u127**
- **2024.q3.3**
- **LPD-36948 Ensuring that the out-of-convention LPD-X commit does not break the release notes**
- **LPD-36948 Add test for generate_release_notes**
- **LPD-36948 Optimize**
- **LPD-36948 Use absolute path instead of relative**
- **LPD-36948 Automatically trigger tests only for user liferay-release**
- **LPD-36948 Disable automatic prepare next temporarily**
- **LPD-36948 SF**
- **LPD-36948 force**
- **LPD-36948 This should still work**
- **2024.q3.4**
- **7.4.13-u128**
- **2024.q3.5**
- **LPD-33239 Remove unused liferay_common.sh**
- **LPD-33239 Move _liferay_common.sh to reuse it out of release folder**
- **LPD-39668 Skip adding internal releases to releases.json**
- **LPD-39668 Add test to cover issue use case**
- **LPD-39668 Optimize**
- **LPD-39668 Only upload releases.json if necessary**
- **LPD-39668 Like this**
- **LPD-39668 Rename**
- **LPD-39668 Sort**
- **LPD-37968 Skip if dependency already exists**
- **LPD-37968 Add test to cover generation of BOM**
- **LPD-37968 Consistency**
- **LPD-37968 SF**
- **LPD-37968 Improve test error output**
- **LPD-37968 Make sure the test name will be displayed**
- **LPD-37968 Fix broken test**
- **LPD-37968 SF**
- **LPD-37968 Clear test results output**
- **LPD-37968 Wordsmith**
- **LPD-37968 Remove unnecessary script import**
- **DOCKER-262 Add jdk21-jdk8 image**
- **DOCKER-262 Switch jar-runner to use jdk21-jdk8**

### CODE REVIEW

**Comment by gamerson** [comment]
Is it at all possible to leave java11 as is and just add on java21?  so it would be a new
jdk21-jdk11-jdk8 image? or is that too much?

**Comment by allen-ziegenfus** [comment]
> Is it at all possible to leave java11 as is and just add on java21? so it would be a new jdk21-jdk11-jdk8 image? or is that too much?

Yeah I started trying to do that but: 

1. I wasn't sure how exactly to do that
2. I thought that might not be looked upon fondly to have so many

But I can look at it again, sure!

**Comment by allen-ziegenfus** [comment]
@gamerson  I found this way of adding jdk21 to the jar runner image.

File: templates/jar-runner/Dockerfile [hunk 1]
```diff
@@ -18,7 +18,7 @@ COPY resources/ /
 
 ENTRYPOINT ["tini", "-v", "--", "/usr/local/bin/liferay_jar_runner_entrypoint.sh"]
 
-ENV JAVA_VERSION=zulu11
+ENV JAVA_VERSION=zulu21
```
Comment (gamerson) [inline templates/jar-runner/Dockerfile]: Could we switch this to keep java11 as default?  But if customers want to switch to java21 they can just put
`JAVA_VERSION=zulu21` in their dockerfile

Comment (allen-ziegenfus) [reply templates/jar-runner/Dockerfile]: sure

Comment (allen-ziegenfus) [reply templates/jar-runner/Dockerfile]: https://github.com/gamerson/liferay-docker/pull/4/commits/af00790f55b1527980ce02e2d35a370a8d5cfcf6

**Comment by allen-ziegenfus** [comment]
https://github.com/brianchandotcom/liferay-docker/pull/755

**Pull request closed without merging**